vignettes/projecting_incidence.Rmd

```diff
@@ -0,0 +1,265 @@
+
+projection_end_day <- max(days_since_index) + projection_window
+
+#' Number of simulations
+sim_rep <- 100
+
+#' Maximum chain size allowed
+chain_threshold <- 1000
+
+```
+
+#### Serial interval
+
+We also assume based on COVID-19 literature that the 
+serial interval, $si$, is lognormal distributed as follows:
+
+$ E[\text{si}] = \ln \left( \dfrac{\mu^2}{(\sqrt{\mu^2 + \sigma^2}} \right)$
+
+$ SD [\text{si}] = \sqrt {\ln \left(1 + \dfrac{\sigma^2}{\mu^2} \right)}$
+
+with $\mu = 4.7$ and standard deviation $\sigma = 2.9$.
+
+```{r input_prep3, message=FALSE}
+mu <- 4.7
+sigma <- 2.9
+
+si_sd <- sqrt(log(1 + (sigma / mu)^2)) # log standard deviation
+si_mean <- log((mu^2) / (sqrt(sigma^2 + mu^2))) # log mean
+
+#' serial interval function
+serial_interval <- function(sample_size) {
+  si <- rlnorm(sample_size, meanlog = si_mean, sdlog = si_sd)
+  return(si)
+}
+```
+
+#### Offspring distribution
+
+We assume an offspring distribution that is distributed as a negative binomial with $R = 2.5$ [@abbott2020] and $k = 0.58$ [@Wang2020]. In this parameterization, R represents the $\mathcal{R_0}$, which is defined as the average number of cases produced by a single individual in an entirely susceptible population. The parameter $k$ represents superspreading, that is, the degree of heterogeneity in transmission by single individuals.
+
+### Simulations
+To summarize the simulation set up, for each of the `r sim_rep` simulations, we want to project cases over a `r projection_window` day period since the last case, assuming that no chain would exceed `r chain_threshold`. 
+
+#### Model assumptions
+
+`chain_sim()` makes the following simplifying assumptions:
+
+1. All cases are observed
+1. There is no reporting delay
+1. Reporting rate is constant through the course of the epidemic
+1. No interventions have been implemented
+1. Population is homogeneous and well-mixed
+
+```{r simulations, message=FALSE}
+set.seed(1234)
+
+
+sim_chain_sizes <- lapply(
+  seq_len(sim_rep),
+  function(sim) {
+    chain_sim(
+      n = length(start_times),
+      offspring = "nbinom",
+      mu = 2.5,
+      size = 0.58,
+      stat = "size",
+      infinite = chain_threshold,
+      serial = serial_interval,
+      t0 = start_times,
+      tf = projection_end_day,
+      tree = TRUE
+    ) %>%
+      mutate(sim = sim)
+  }
+)
+
+sim_output <- do.call(rbind, sim_chain_sizes)
+
+head(sim_output)
+```
+
+From the simulated data, we count the median daily cases across 
+all simulations and overlay that over a plot of all the projections through time.
+
+#### Post-processing
+
+```{r post_processing}
+index_date <- min(covid19_sa$date)
+
+# Daily number of cases for each simulation
+incidence_ts <- sim_output %>%
+  mutate(day = ceiling(time)) %>%
+  group_by(sim, day) %>%
+  summarise(cases = n()) %>%
+  ungroup()
+
+# Add dates
+incidence_ts <- incidence_ts %>%
+  group_by(sim) %>%
+  mutate(date = index_date + (0:(n() - 1))) %>%
```

Comment:
Suggest using instead `date = index_date + days(seq(0, n() - 1))` to make it clearer that you're adding days to the minimum date. In general, avoid ranges specified `0:n`

Reply:
Thanks, that's insightful.